scripts/release.py

```diff
@@ -0,0 +1,199 @@
+import json
+import os
+import re
+import subprocess
+
+from dotenv import load_dotenv
+from github import Github
+
+
+"""This release notes script is built to create a release notes draft for release candidates, patches, and minor releases.
+
+Setup:
+1. Create a `.env` file in the scripts directory.
+2. Create Github token: https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token#creating-a-personal-access-token-classic
+3. Give the Github token repo, user, audit_log, and project permissions.
+2. Add `export GH_TOKEN=<github token>` to the `.env` file.
+
+Usage:
+Required:
+    BASE - The base branch you are building your release candidate, patch, or minor release off of.
+        If this is a rc1, then just specify the branch you'll create after the release is published. e.g. BASE=1.9
+
+Optional:
+    RC - Whether or not this is a release candidate. e.g. RC=1 or RC=0
+    PATCH - Whether or not this a patch release. e.g. PATCH=1 or PATCH=0
+    
+Examples:
+Generate release notes for next release candidate version of 1.11: `BASE=1.11 RC=1 python release.py`
+
+Generate release notes for next patch version of 1.13: `BASE=1.13 PATCH=1 python release.py`
+
+Generate release notes for the 1.15 release: `BASE=1.15 python release.py`
+"""
+
+home_dir = os.path.expanduser("~")
+env_loaded = load_dotenv(dotenv_path="%s/.env" % home_dir)
+if not env_loaded:
+    raise ValueError("No envars were loaded from .env file. Please follow the instructions in the script.")
+
+
+def create_release_draft():
+    #  Figure out the version of the library that you’re working on releasing grabbed with VERSION envar
+    base = os.getenv("BASE")
+    gh_token = os.getenv("GH_TOKEN")
+    rc = bool(os.getenv("RC"))
+    patch = bool(os.getenv("PATCH"))
+
+    if base is None:
+        raise ValueError("Need to specify the base version with envar e.g. BASE=1.10.0")
+
+    # make sure we're up to date
+    subprocess.run("git fetch", shell=True, cwd=os.pardir)
+
+    # setup gh
+    g = Github(gh_token)
+    dd_repo = g.get_repo(full_name_or_id="DataDog/dd-trace-py")
+
+    if rc:
+        # figure out the rc version we want
+        search = "v%s.0\.?rc((\d+$))" % base
+        tags = dd_repo.get_tags()
+        rc_version = 0
+        for tag in tags:
+            try:
+                rc_num = re.findall(search, tag.name)[0][0]
+                rc_num = int(rc_num)
+            except (IndexError, ValueError, TypeError):
+                continue
+            if rc_num > rc_version:
+                rc_version = rc_num
```

Comment:
Confused on what this loop is doing. Is it finding the most recent RC tag for the given minor version?

Reply:
I was primarily confused by what `rc_version` means, I am suspecting that this is the most recent RC version instead of the RC version we want to generate release notes for. Can we clarify this? For example renaming `rc_version` to `most_recent_rc` and making `rc_version = most_recent_rc + 1`

Reply:
Yes correct, it's finding the rc version we want to use by figuring out what the most recent rc of that minor version was and adding 1 to it. Yeah I'll clarify the variable names more. Thanks!